pkg/cmd/cli/cmd/debug.go

```diff
@@ -531,17 +532,24 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*kap
 		if pod.Labels == nil {
 			pod.Labels = make(map[string]string)
 		}
-		pod.Labels[debugPodLabelName] = pod.Name
 	} else {
-		pod.Labels = map[string]string{debugPodLabelName: pod.Name}
+		pod.Labels = map[string]string{}
 	}
 	// always clear the NodeName
 	pod.Spec.NodeName = o.NodeName
 
 	pod.ResourceVersion = ""
 	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
-	// TODO: shorten segments, make incrementing?
-	pod.Name = fmt.Sprintf("%s-debug", pod.Name)
+	// shorten segments to handle long names and names with bad characters
```

Comment:
No no, github.com/openshift/origin/pkg/generate/app.  This is a generally
useful function for turning a string into a pod name.  That's stuff we
should have / be using / be reusing in generation (we already have this for
services, which used to be more restrictive).

On Mon, May 9, 2016 at 3:09 PM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/cli/cmd/debug.go
> <https://github.com/openshift/origin/pull/8742#discussion_r62555225>:
>
> >  	}
> >  	// always clear the NodeName
> >  	pod.Spec.NodeName = o.NodeName
> >
> >  	pod.ResourceVersion = ""
> >  	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
> > -	// TODO: shorten segments, make incrementing?
> > -	pod.Name = fmt.Sprintf("%s-debug", pod.Name)
> > +	// shorten segments to handle long names and names with bad characters
>
> Can you move this to a function in the generate libraries please?
>
> You mean down here:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/api/generate.go#L59
> ? How about I conform to the interface and upstream later if something else
> ever needs it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8742/files/d349937a294f3a4d7908882459ba8b2ee15706d8#r62555225>
>